### ISSUE TITLE
[BUG] pf flow serve is broken in promptflow==1.10.0 (NotADirectoryError)

### ISSUE THREAD

**Issue by asos-oliverfrost** [microsoft/promptflow#3058]
> Clearly state that you are going to take on implementing it, if that's the case. You can request that the issue be assigned to you.

**I'm happy to implement this** seems straightforward (possible solution provided below, please confirm if this is the expected behaviour)

**Describe the bug**

Serving the interactive chat UI appears to have broken. Flask app 'promptflow.core._serving.app' does not start successfully, resulting in an exception and the local app server doesn't open in a browser.

**How To Reproduce the bug**

Using the following `flow.dag.yaml`:

```yaml
$schema: https://azuremlschemas.azureedge.net/promptflow/latest/Flow.schema.json
inputs:
  question:
    type: string
    is_chat_input: true
nodes:
- name: parrot
  type: python
  source:
    type: code
    path: ./parrot.py
  inputs:
    question: ${inputs.question}
outputs:
  answer:
    type: string
    reference: ${parrot.output}
    is_chat_output: true
```

and `parrot.py` (both files in the same source directory):

```python
from promptflow.core import tool

@tool
def print_hi(question: str):
    return question.upper()
```

To reproduce the error, run any of the following `pf` commands below:

```shell
pf flow serve --source ./
pf flow serve --source .
pf flow serve --source flow.dag.yaml
```

Exception raised:

```text
  File "/Users/oliver.frost/.pyenv/versions/3.11.4/lib/python3.11/site-packages/promptflow/_sdk/_utils/serve_utils.py", line 112, in serve_python_flow
    os.chdir(flow_dir)
NotADirectoryError: [Errno 20] Not a directory: '/Users/oliver.frost/PycharmProjects/pf-bug-example/flow.dag.yaml'
```

Issue resolves when you adjust `_resolve_python_flow_additional_includes()` to return the parent of the source instead, included below:

```python
def _resolve_python_flow_additional_includes(source) -> Path:
    # Resolve flow additional includes
    from promptflow.client import load_flow

    flow = load_flow(source)
    from promptflow._sdk.operations import FlowOperations

    with FlowOperations._resolve_additional_includes(flow.path) as resolved_flow_path:
        if resolved_flow_path == flow.path:
            # HERE
            return source.parent
```

**Expected behavior**

Flask app 'promptflow.core._serving.app' serves successfully and a browser window opens:

<img width="910" alt="Screenshot 2024-04-28 at 09 56 54" src="https://github.com/microsoft/promptflow/assets/87647306/bde63e16-2a42-48b5-8b81-84370093a364">

**Screenshots**

<img width="1057" alt="Screenshot 2024-04-28 at 09 48 59" src="https://github.com/microsoft/promptflow/assets/87647306/21640aaa-14e9-4fad-9c2f-080d704f5a81">
Above: the reproducible example in PyCharm.

<img width="1217" alt="Screenshot 2024-04-28 at 09 54 00" src="https://github.com/microsoft/promptflow/assets/87647306/cdc43447-e1f1-4548-ac63-4b64a9fa972f">
^ above: same error reproduced in Python 3.9.x

**Running Information(please complete the following information):**

 - Promptflow == 1.10.0
 - Operating System: Mac OS X 14.4.1 (23E224)
 - Python Version 3.11.4

**Comment by brynn-code** [comment]
@asos-oliverfrost Thanks for reporting this! We will fix this asap in the postfix 1.10.1.

**Comment by asos-oliverfrost** [comment]
Thanks for the reply! I was hoping to take a crack at this, as we've worked a lot with promptflow and I was hoping to become more active as a contributor :)

**Comment by brynn-code** [comment]
Haha, thank you, we are fixing this here #3070 because this related to some code refine works not so long ago, and this breaks 1.10.0 so it's a little hurry to ship the fix in 1.10.1, I believe you've been our contributor by reporting this, really appreciate it, and contribute code changes will not be too far for you 😆, thanks again. @asos-oliverfrost

**Comment by brynn-code** [comment]
Close as fixed.
